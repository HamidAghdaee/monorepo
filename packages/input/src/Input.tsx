import * as React from 'react';
import styled from 'styled-components';

export interface LabelProps {
  labelWidth?: number;
}

export interface InputWithLabelProps extends LabelProps {
  id?: string;
  label?: string;
}

export interface InputWithoutLabelProps extends LabelProps {
  id: string;
  label: string;
}

export type InputLabelProps = InputWithLabelProps | InputWithoutLabelProps;

export interface InputProps {
  name?: string;
  type?: string;
}

const Wrapper = styled.div`
  display: flex;
  margin: 10px;
`;

const Label = styled<LabelProps, 'label'>('label')`
  margin-right: 10px;
  font-weight: bold;
  width: ${({ labelWidth = 120 }) => labelWidth}px;
`;

const NativeInput = styled.input`
  width: 100%;
`;

export const Input: React.SFC<InputProps & InputLabelProps> = ({
  label,
  id,
  labelWidth,
  ...rest
}) => (
  <Wrapper>
    {label && (
      <Label labelWidth={labelWidth} htmlFor={id}>
        {label}:
      </Label>
    )}
    <NativeInput id={id} {...rest} />
  </Wrapper>
);

Input.defaultProps = {
  type: 'text',
};
