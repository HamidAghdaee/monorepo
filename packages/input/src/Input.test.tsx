import { mount } from 'enzyme';
import * as React from 'react';

import { Input } from './Input';

describe('Input', () => {
  test('should match snapshot and styles for default props', () => {
    expect(mount(<Input />)).toMatchSnapshot();
  });

  test('should match snapshot with label', () => {
    expect(mount(<Input id="test" label="Name" />)).toMatchSnapshot();
  });
});
