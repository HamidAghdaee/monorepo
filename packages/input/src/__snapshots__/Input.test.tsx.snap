// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot and styles for default props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 10px;
}

.c1 {
  width: 100%;
}

<Component
  type="text"
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.input
        type="text"
      >
        <input
          className="c1"
          type="text"
        />
      </styled.input>
    </div>
  </styled.div>
</Component>
`;

exports[`Input should match snapshot with label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 10px;
}

.c1 {
  margin-right: 10px;
  font-weight: bold;
  width: 120px;
}

.c2 {
  width: 100%;
}

<Component
  id="test"
  label="Name"
  type="text"
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.label
        htmlFor="test"
      >
        <label
          className="c1"
          htmlFor="test"
        >
          Name
          :
        </label>
      </styled.label>
      <styled.input
        id="test"
        type="text"
      >
        <input
          className="c2"
          id="test"
          type="text"
        />
      </styled.input>
    </div>
  </styled.div>
</Component>
`;
